sdk/storage/Azure.Storage.Files.DataLake/src/Models/PathPermissions.cs

```diff
@@ -0,0 +1,213 @@
+        /// or groups.
+        /// </summary>
+        public bool ExtendedInfoInAcl { get; set; }
+
+        /// <summary>
+        /// Internal empty constructor.
+        /// </summary>
+        public PathPermissions() { }
+
+        /// <summary>
+        /// Public constructor.
+        /// </summary>
+        /// <param name="owner">The path owner's permissions.</param>
+        /// <param name="group">The path group's permissions.</param>
+        /// <param name="other">Permissions for other users.</param>
+        /// <param name="stickyBit">If sticky bit is enabled</param>
+        /// <param name="extendedInfoInAcl">If there is extended info in the ACL</param>
+        public PathPermissions(
+            RolePermissions owner,
+            RolePermissions group,
+            RolePermissions other,
+            bool stickyBit = false,
+            bool extendedInfoInAcl = false)
+        {
+            Owner = owner;
+            Group = group;
+            Other = other;
+            StickyBit = stickyBit;
+            ExtendedInfoInAcl = extendedInfoInAcl;
+        }
+
+        /// <summary>
+        /// Parses a string in octal format to PathPermissions.
+        /// </summary>
+        /// <param name="s">Octal string to parse.</param>
+        /// <returns><see cref="PathPermissions"/>.</returns>
+        public static PathPermissions ParseOctalPermissions(string s)
+        {
+            if (s == null)
+            {
+                return null;
+            }
+
+            if (s.Length != 4)
+            {
+                throw new ArgumentException($"{nameof(s)} must be 4 characters");
+            }
+
+            var pathPermissions = new PathPermissions();
+
+            if (s[0] == '0')
+            {
+                pathPermissions.StickyBit = false;
+            }
+            else if (s[0] == '1')
+            {
+                pathPermissions.StickyBit = true;
+            }
+            else
+            {
+                throw new ArgumentException($"First digit of {nameof(s)} must be 0 or 1");
+            }
+
+            pathPermissions.Owner = PathAccessControlExtensions.ParseOctalRolePermissions(s[1]);
+            pathPermissions.Group = PathAccessControlExtensions.ParseOctalRolePermissions(s[2]);
+            pathPermissions.Other = PathAccessControlExtensions.ParseOctalRolePermissions(s[3]);
+
+            return pathPermissions;
+        }
+
+        /// <summary>
+        /// Parses a symbolic string to PathPermissions.
+        /// </summary>
+        /// <param name="s">String to parse.</param>
+        /// <returns><see cref="PathPermissions"/>.</returns>
+        public static PathPermissions ParseSymbolicPermissions(string s)
+        {
+            if (s == null)
+            {
+                return null;
+            }
+
+            if (s.Length != 9 && s.Length != 10)
+            {
+                throw new ArgumentException($"{nameof(s)} must be 9 or 10 characters");
```

Comment:
Move to DataLakeErrors.cs and include the actual permissions string you're trying to parse

Reply:
Add string, see my comment above.

Reply:
fixed.